tests/phpunit/api/v4/Action/BasicCustomFieldTest.php

```diff
@@ -272,4 +273,39 @@ public function testRelationshipCacheCustomFields() {
     $this->assertEquals('Buddy', $results[0]["$cgName.PetName"]);
   }
 
+  /**
+   * Some types are creating a dummy option group even if we don't have
+   * any option values.
+   * @throws \API_Exception
+   */
+  public function testUndesiredOptionGroupCreation(): void {
+    $optionGroupCount = OptionGroup::get(FALSE)->selectRowCount()->execute()->count();
+
+    $customGroup = CustomGroup::create(FALSE)
+      ->addValue('name', 'MyIndividualFields')
+      ->addValue('extends', 'Individual')
+      ->execute()
+      ->first();
+
+    // This one doesn't make sense to have an option group.
+    CustomField::create(FALSE)
+      ->addValue('label', 'FavColor')
+      ->addValue('custom_group_id', $customGroup['id'])
+      ->addValue('html_type', 'Number')
+      ->addValue('data_type', 'Money')
+      ->execute();
+
+    // This one might be ok if we planned to then use the autocreated option
+    // group, but if we go on to create our own after then we have an extra
+    // unused group.
```

Comment:
Yea I would say no this one is not ok. Because custom fields can re-use existing option groups. They can also have options supplied via hook.
I agree with the assertion at the end of this test that no new optionGroups should have been created.